### PR TITLE
Add test for containerapp up and fix bug

### PR DESCRIPTION
**Pull request by Greedygre**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az containerapp up

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_up.py [hunk 1]
```diff
@@ -32,3 +32,29 @@ def test_containerapp_up_image_e2e(self, resource_group):
...
+        self.cmd(f'aks create --resource-group {resource_group} --name {aks_name} --enable-aad --generate-ssh-keys')
+        self.cmd(f'aks get-credentials --resource-group {resource_group} --name {aks_name} --overwrite-existing --admin')
+        self.cmd(f'connectedk8s connect --resource-group {resource_group} --name {connected_cluster_name}')
+        connected_cluster = self.cmd(f'az connectedk8s show --resource-group {resource_group} --name {connected_cluster_name}').get_output_in_json()
+        connected_cluster_id = connected_cluster['id']
+        app_name = 'mycontainerapp'
+        app_rg = 'my-containerapp-rg'
+        image = 'mcr.microsoft.com/azuredocs/containerapps-helloworld:latest'
+
+        self.cmd(f'containerapp up -n {app_name} -g {app_rg} --connected-cluster-id {connected_cluster_id} --image {image}')
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_up.py]: could we add test cases for all scenarios we designed for arc?

File: src/containerapp/azext_containerapp/_utils.py [hunk 2]
```diff
@@ -1594,17 +1592,17 @@ def create_extension(cmd, extension_name='containerapps-ext', connected_cluster_
     cluster_name = parsed_extension.get("name")
 
     e = models.Extension()
+    e.identity = models.Identity(type="SystemAssigned")
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_utils.py:1595]: why do we need the identity ?

File: src/containerapp/setup.py [hunk 3]
```diff
@@ -39,7 +39,7 @@
 # TODO: Add any additional SDK dependencies here
 DEPENDENCIES = [
     'azure-cli-core',
-    'azure-mgmt-kubernetesconfiguration==2.0.0',
+    'azure-mgmt-kubernetesconfiguration==3.0.0b1',
```
Comment (Juliehzl) [inline src/containerapp/setup.py]: why do you need to use beta version sdk?

File: src/containerapp/azext_containerapp/_up_utils.py [hunk 4]
```diff
@@ -707,6 +707,8 @@ def _get_app_env_and_group(
             matched_apps = [c for c in matched_apps if (c.get("extendedLocation") and c["extendedLocation"]["name"].lower() == env.custom_location_id.lower())]
         elif env.resource_type:
             matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["environmentId"])["resource_type"].lower() == env.resource_type.lower()]
+        if custom_location.connected_cluster_id:
+            matched_apps = _filter_containerapps_by_connected_cluster_id(cmd, matched_apps, custom_location.connected_cluster_id)
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_up_utils.py:711]: is it possible to have two custom location in different connected clusters?

Comment (Greedygre) [reply src/containerapp/azext_containerapp/_up_utils.py:711]: Yes. If user input custom-location, we filter by custom location first.

File: src/containerapp/azext_containerapp/_utils.py (hunk 2 above)
Comment (Greedygre) [reply src/containerapp/azext_containerapp/_utils.py:1595]: Keep consistent with 'az k8s-extension create' command.

File: src/containerapp/setup.py (hunk 3 above)
Comment (Greedygre) [reply src/containerapp/setup.py]: revert to 2.0.0

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_up.py [hunk 5]
```diff
@@ -32,3 +32,82 @@ def test_containerapp_up_image_e2e(self, resource_group):
...
+        # -n {appname} --custom-location --environment {env_name}
+        self.cmd(f'containerapp up -n {app_name} -g {resource_group} --custom-location {custom_location_id} --environment {env_id} --image {image}')
+        self._validate_app(resource_group, app_name)
+
+        # -n {appname} --custom-location --environment {env_id}
+        self.cmd(f'containerapp up -n {app_name} -g {resource_group} --custom-location {custom_location_id} --environment {env_name} --image {image}')
+        self._validate_app(resource_group, app_name)
+
+    def _validate_app(self, resource_group, app_name):
+        app = self.cmd(f"containerapp show -g {resource_group} -n {app_name}").get_output_in_json()
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_up.py:111]: validate custom location value

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**